pkg/volume/hostpath/host_path.go

```diff
@@ -483,8 +484,16 @@ func makeDir(pathname string) error {
 }
 
 // makeFile creates an empty file.
+// If the directory where the file is located does not exist, create the directory first
 // If pathname already exists, whether a file or directory, no error is returned.
 func makeFile(pathname string) error {
+	path, _ := filepath.Split(pathname)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```

Comment:
There's high levels of variable shadowing here - for the avoidance of doubt, I would rename this to `statErr` and keep the makeDir result as `err`.

Reply:
Great suggestions, updated, thanks a lot